src/background/vvppManager.ts

```diff
@@ -120,14 +117,30 @@ export class VvppManager {
     };
   }
 
-  async processDeferredProcesses() {
-    const { vvppEngineDir } = this.getEngineDirPaths();
+  async load(vvppPath: string) {
+    const { outputDir, manifest } = await this.extractVvpp(vvppPath);
+    let willMove = false;
+    const dirName = this.toValidDirName(manifest);
+    const engineDirectory = path.join(this.vvppEngineDir, dirName);
+    for (const dir of await fs.promises.readdir(this.vvppEngineDir)) {
+      if (!this.isEngineDirName(dir, manifest)) {
+        continue;
+      }
+      this.markWillMove(outputDir, dirName);
+      willMove = true;
+    }
+    if (!willMove) {
+      await moveFile(outputDir, engineDirectory);
+    }
+  }
```

Comment:
以前のエンジンがあれば上書きを予約、なければmove、という処理だと思うのですが、ちょっと見通しが立ちにくくなっていそうです。
「以前のエンジンディレクトリ名」という状態変数を作るようにするとわかりやすいと思います。
```suggestion
  async load(vvppPath: string) {
    const { outputDir, manifest } = await this.extractVvpp(vvppPath);
    let willMove = false;
    const dirName = this.toValidDirName(manifest);
    const engineDirectory = path.join(this.vvppEngineDir, dirName);
    const oldEngineDirName = await fs.promises.readdir(this.vvppEngineDir).find((dir) => {
      return this.isEngineDirName(dir, manifest);
    })
    if (oldEngineDirName) {
      this.markWillMove(outputDir, dirName);
    }
    else {
      await moveFile(outputDir, engineDirectory);
    }
  }
```
もしくは処理の流れをloadのドキュメントコメントに書くとかでも大丈夫かなと！

`!willMove`のときに`moveFile`する、というのが混乱しそうです。
`willMove`を全部`willReplace`に変えていくのはどうでしょう。